### PR TITLE
Add support for generated columns

### PR DESCRIPTION
**Pull request by tooboredtocode**
## New Features

- [x] Generated columns

## Breaking Changes

This change is backwards compatible

### CODE REVIEW

**Review by ikrivosheev — COMMENTED** [review]
@tooboredtocode hello! Thank you for the PR! Can you add a link to documentation. I have never heard about this column type...

**Comment by ikrivosheev** [comment]
> @tooboredtocode hello! Thank you for the PR! Can you add a link to documentation. I have never heard about this column type...

I find an answer. Discussion was on discord: https://discord.com/channels/873880840487206962/900758332187484160/1063039879518359562

**Comment by tooboredtocode** [comment]
Here are the relevant documentations:
https://www.postgresql.org/docs/current/ddl-generated-columns.html
https://dev.mysql.com/doc/refman/5.7/en/create-table-generated-columns.html
https://www.sqlite.org/gencol.html

**Review by ikrivosheev — APPROVED** [review]
@tooboredtocode hello! Sorry for the delay, I had a vacation. LGTM!

**Comment by ikrivosheev** [comment]
@tooboredtocode could you resolve conflicts?

**Comment by tooboredtocode** [comment]
@ikrivosheev done 👍

**Comment by tyt2y3** [comment]
Thank you. Can we add a doc test?

**Comment by tooboredtocode** [comment]
Where could they be added, i did not find any similar tests where the new tests could fit in.

File: src/table/column.rs [hunk 1]
```diff
@@ -584,6 +585,18 @@ impl ColumnDef {
         self
     }
 
+    /// Sets the column as generated with SimpleExpr
+    pub fn generated<T>(&mut self, expr: T, stored: bool) -> &mut Self
```
Comment (tyt2y3) [inline src/table/column.rs:589]: May be a doc test like above ? https://github.com/tooboredtocode/sea-query/blob/b6adc4b0f18fa24fafa6ac543bf49a4da190758e/src/table/column.rs#L246

**Pull request merged**